### PR TITLE
To spark1 4

### PR DESCRIPTION
**Pull request by josegom**
I have changed the code to adapt this project to  Apache Spark SQL 1.4.

### CODE REVIEW

**Comment by yzhou2001** [comment]
@josegom Thanks for the 2 PRs. They are important work I believe. In particular, the adoption of Spark 1.4 is something we have been pondering over past weeks but haven't found bandwidth to do so yet. I'll review them later.

Please also note that for historical reasons, the mainstream development work is actually at the "hbase" branch of https://github.com/Huawei-Spark/spark, and not here. Later the work will be merged here. In the future (probably a couple of months later) we will solely use this repo as the dev point.

**Comment by yzhou2001** [comment]
@josegom I realize that the adoption of 1.4 is better to be performed after the merge from the "hbase" branch to this repo which is planned to be after the branch is stable, probably in around one or two months. Would you mind to create a PR then for the adoption of 1.4?

For your commit of 9375503, could you summarize what are the key changes and your thoughts behind the changes?  Thanks.

**Comment by josegom** [comment]
I don't have any problem to make the PR when this repo will be stable but I am using Apache Spark SQL 1.4 and Apache Hbase now. For this reason I need to use my PR and I would like to know if there exist any problem to use it or to publish it (for example in a public maven repository) until the final version will be released.

On the other hand I would like to know if you are planning the continuous adaptatation to Apache Spark, because  it's probably that a new Spark version will be published in three or four months.

Finally the main changes in the PR have been as follows:
- Turn NativeType into AtomicType
- Turn JvmType into InternalType
- Two or three methods with new parameters.
- Turn apply methods into specific methods
  
  In my opinion the adaptation is easy and the impact to Hbase is only cosmetics.
  
  Thanks.

**Comment by yzhou2001** [comment]
@josegom Yes, we will adapt to the latest Spark release. During the past few months, due to the hectic activity on our dev branch, we have not been able to follow the practice yet. Good news is, thanks to our boosted efforts, we are very close (in a few days hopefully) to merge the dev branch to this place and support 1.4. You're welcome to follow the progress here, test it and give out comments/fixes/..., etc.

**Comment by yzhou2001** [comment]
We have created a "Stratio-ToSpark1_4" branch for the merge work. We'll appreciate if you can test and even contribute PRs against the branch. In particular, Python and Java API tests have not been executed.

We are rushing to conclude the work as soon as possible, but feel short of hands at the moment. 

Thanks.

**Comment by yzhou2001** [comment]
@josegom my last comments were addressed to you.

**Comment by josegom** [comment]
I close this PR and test the branch. If I find any bugs or improvements I will report it in stratio-ToSpark1_4 branch.

Thanks.

**Comment by xinyunh** [comment]
@josegom Thanks！
